src/library.c

```diff
@@ -0,0 +1,850 @@
+      mfi->media_kind = MEDIA_KIND_TVSHOW;  /* tv show */
+
+      /* Default to artist = series_name */
+      if (mfi->artist && strlen(mfi->artist) == 0)
+	{
+	  free(mfi->artist);
+	  mfi->artist = NULL;
+	}
+
+      if (!mfi->artist)
+	mfi->artist = strdup(mfi->tv_series_name);
+
+      /* Default to album = "<series_name>, Season <season_num>" */
+      if (mfi->album && strlen(mfi->album) == 0)
+	{
+	  free(mfi->album);
+	  mfi->album = NULL;
+	}
+
+      if (!mfi->album)
+	{
+	  len = snprintf(NULL, 0, "%s, Season %d", mfi->tv_series_name, mfi->tv_season_num);
+
+	  mfi->album = (char *)malloc(len + 1);
+	  if (mfi->album)
+	    sprintf(mfi->album, "%s, Season %d", mfi->tv_series_name, mfi->tv_season_num);
+	}
+    }
+
+  /* Check the 4 top-tags are filled */
+  if (!mfi->artist)
+    mfi->artist = strdup("Unknown artist");
+  if (!mfi->album)
+    mfi->album = strdup("Unknown album");
+  if (!mfi->genre)
+    mfi->genre = strdup("Unknown genre");
+  if (!mfi->title)
+    {
+      /* fname is left untouched by unicode_fixup_mfi() for
+       * obvious reasons, so ensure it is proper UTF-8
+       */
+      mfi->title = unicode_fixup_string(mfi->fname, "ascii");
+      if (mfi->title == mfi->fname)
+	mfi->title = strdup(mfi->fname);
+    }
+
+  /* Ensure sort tags are filled, manipulated and normalized */
+  sort_tag_create(&mfi->artist_sort, mfi->artist);
+  sort_tag_create(&mfi->album_sort, mfi->album);
+  sort_tag_create(&mfi->title_sort, mfi->title);
+
+  /* We need to set album_artist according to media type and config */
+  if (mfi->compilation)          /* Compilation */
+    {
+      lib = cfg_getsec(cfg, "library");
+      ca = cfg_getstr(lib, "compilation_artist");
+      if (ca && mfi->album_artist)
+	{
+	  free(mfi->album_artist);
+	  mfi->album_artist = strdup(ca);
+	}
+      else if (ca && !mfi->album_artist)
+	{
+	  mfi->album_artist = strdup(ca);
+	}
+      else if (!ca && !mfi->album_artist)
+	{
+	  mfi->album_artist = strdup("");
+	  mfi->album_artist_sort = strdup("");
+	}
+    }
+  else if (mfi->media_kind == MEDIA_KIND_PODCAST) /* Podcast */
+    {
+      if (mfi->album_artist)
+	free(mfi->album_artist);
+      mfi->album_artist = strdup("");
+      mfi->album_artist_sort = strdup("");
+    }
+  else if (!mfi->album_artist)   /* Regular media without album_artist */
+    {
+      mfi->album_artist = strdup(mfi->artist);
+    }
+
+  if (!mfi->album_artist_sort && (strcmp(mfi->album_artist, mfi->artist) == 0))
+    mfi->album_artist_sort = strdup(mfi->artist_sort);
+  else
+    sort_tag_create(&mfi->album_artist_sort, mfi->album_artist);
+
+  /* Composer is not one of our mandatory tags, so take extra care */
+  if (mfi->composer_sort || mfi->composer)
+    sort_tag_create(&mfi->composer_sort, mfi->composer);
+}
+
+void
+library_process_media(const char *path, time_t mtime, off_t size, enum data_kind data_kind, enum media_kind force_media_kind, bool force_compilation, struct media_file_info *external_mfi, int dir_id)
```

Comment:
I notice you started using bool a few places in this pr. I'm all for that, but I'm wondering how standard bool is these days? Its been years since I last attempted bool, so not sure about the state of it.